program-bpf-rust/do.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+cd "$(dirname "$0")"
+
+usage() {
+    echo ""
+    echo "  Usage: do.sh action <project>"
+    echo ""
+    echo "  If relative_project_path is ommitted then action will"
+    echo "  be performed on all projects"
+    echo ""
+    echo "  Supported actions:"
+    echo "    build"
+    echo "    clean"
+    echo "    test"
+    echo "    clippy"
+    echo "    fmt"
+    echo ""
+}
+
+perform_action() {
+    set -e
+    case "$1" in
+    build)
+         ../node_modules/@solana/web3.js/bpf-sdk/rust/build.sh "$PWD"
+         mkdir -p ../dist/program
+         cp ./target/bpfel-unknown-unknown/release/tictactoe.so ../dist/program
+    ;;
+    clean)
+         ../node_modules/@solana/web3.js/bpf-sdk/rust/clean.sh "$PWD"
+    ;;
+    test)
+        (
+            cd "$2"
+            echo "test $2"
+            cargo +nightly test
+        )
+    ;;
+    clippy)
+        (
+            echo "clippy $2"
+            cargo +nightly clippy
+        )
+    ;;
+    fmt)
+        (
+            echo "formatting $2"
+            cargo fmt
+        )
+    ;;
```

Comment:
nit: use consistent `;;` indent (see help command for the other style, which I personally like more).